src/collisionAvoidence.cpp

```diff
@@ -4,44 +4,54 @@
 
 class SubAndPub
 {
-public:
-	SubAndPub()
-	{
-		pub = n.advertise<geometry_msgs::Twist>("/cmd_vel",10);
-
-		sub = n.subscribe("/scan",10, &SubAndPub::cmdVelCallback, this);
-	}
-	void cmdVelCallback(const sensor_msgs::LaserScan::ConstPtr& scan_msg)
-	{
-		geometry_msgs::Twist velMsg;
-		if(scan_msg->ranges[540] < 2)
-		{
-			velMsg.linear.x = 0.0;
-			velMsg.angular.z = 1.8;
-		}
-		else
-		{
-			velMsg.linear.x = 1.5;
-			velMsg.angular.z = 1.8;
-		}
-		pub.publish(velMsg);
-
-	}
-private:
-	ros::NodeHandle n;
-	ros::Publisher pub;
-	ros::Subscriber sub;
+  public:
+  SubAndPub() {
+      pub = n.advertise<geometry_msgs::Twist>("/cmd_vel",10);
+
+      sub = n.subscribe("/scan",10, &SubAndPub::cmdVelCallback, this);
+  }
+  void cmdVelCallback(const sensor_msgs::LaserScan::ConstPtr& scan_msg);
+
+  private:
+  ros::NodeHandle n;
+  ros::Publisher pub;
+  ros::Subscriber sub;
 };
-	int main(int argc, char **argv)
-	{
-
-		ros::init(argc, argv, "collisionAvoidence");
-
-		SubAndPub substerAndPubster;
-		ros::spin();
-
-		return 0;
 
+void SubAndPub::cmdVelCallback(const sensor_msgs::LaserScan::ConstPtr& scan_msg)
+{
+  geometry_msgs::Twist velMsg;
+  bool shouldTurn = false;
+  /* if(scan_msg->ranges[540] < 1.0 &&
+     scan_msg->ranges[539] < 1.0 &&
+     scan_msg->ranges[538] < 1.0 &&
+     scan_msg->ranges[541] < 1.0) {
+  */
+  for (int i = scan_msg->ranges.size() / 4; i < (scan_msg->ranges.size() - (scan_msg->ranges.size() / 4)); ++i) {
+      if (scan_msg->ranges[i] < 1) {
+          shouldTurn = true;
+          break;
+      }
+  }
+  if(shouldTurn){
+    ROS_INFO("TURN!!!");
+    velMsg.linear.x = 0.0;
+    velMsg.angular.z = .3;
```

Comment:
It would be nice to pick these parameters up from the command line, or, better yet, the ros parameter server. We'll fix that in a different PR.